### PR TITLE
HSEARCH-2199 Allow to define CharFilters with the programmatic API

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

File: orm/src/test/java/org/hibernate/search/test/query/dsl/DSLTest.java [hunk 1]
```diff
@@ -1038,6 +1040,40 @@ public void testUseMatchQueryOnEmbeddedNumericIdCreatesTermQuery() throws Except
 		);
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HSEARCH-2199")
+	@Category(SkipOnElasticsearch.class)
```
Comment (gunnarmorling) [inline orm/src/test/java/org/hibernate/search/test/query/dsl/DSLTest.java]: Can you add a quick note why it's excluded on ES.

File: orm/src/test/java/org/hibernate/search/test/query/dsl/DSLTest.java [hunk 2]
```diff
@@ -1157,7 +1196,10 @@ public SearchMapping build() {
 					.filter( StopFilterFactory.class )
 					.filter( NGramFilterFactory.class )
 					.param( "minGramSize", "3" )
-					.param( "maxGramSize", "3" );
+					.param( "maxGramSize", "3" )
+					.analyzerDef( "htmlStrip", StandardTokenizerFactory.class )
+					.charFilter( HTMLStripCharFilterFactory.class )
```
Comment (gunnarmorling) [inline orm/src/test/java/org/hibernate/search/test/query/dsl/DSLTest.java]: Any factory with parameters we could use for testing?

**Comment by gunnarmorling** [comment]
@gsmet, looks good to me overall. Added two minor remarks inline. Good to go otherwise.

**Comment by gsmet** [comment]
@gunnarmorling done!

I took the liberty to add more comments in DSLTest while I was at it.

**Comment by gunnarmorling** [comment]
Merged; Thanks, @gsmet.

**Pull request merged**